### DISCUSSION TITLE
Question about the Google IDToken Mutator integration

### DISCUSSION THREAD

**Discussion by shikanime** [ory/oathkeeper#959] [Q&A] [answered]
Hello,

I was wondering if there are any plans for first class citizens cloud platform integration since https://github.com/ory/oathkeeper/issues/177? For example, generating an idtoken from the current compute service account, which I reimplemented using hydrator mimicking the id_token configuration API (https://github.com/infinity-blackhole/oathkeeper-google/blob/800bb85055214c88da505daa73dd65b9efaee16c/main.go).

If contributions are welcome, I can take a look at Google Cloud.

**Comment by aeneasr** [comment, marked as answer]
That sounds quite interesting, I think such a feature would first need a design document for a general architecture and feature scope (e.g. generating service account JWTs), as well as one design doc for each cloud platform. But in general, I think this is a quite interesting approach that could also be combined with the various KMS systems these cloud providers offer.
